### PR TITLE
Feature/SK-1081 | Use stores in Combiner + ModelPredict

### PR DESCRIPTION
**Pull request by Wrede**
This pull request includes extensive changes to refactor the terminology from "inference" to "prediction" across the codebase, along starting using stores instead of mongostatestore.py in combiner. Adds ModelPredict message and SendModelPrediction method in proto to allow for Inference workflows. Also, keepalive arguments/options have been added for grpc channels and server. Below are the most important changes:

Breaking change:
endpoint /api/v1/infer -> /api/v1/predict

### Terminology Refactor (Inference to Prediction)

* [`.ci/tests/examples/prediction_test.py`](diffhunk://#diff-ed376aa550f406ce7c8755ac93a996f5c8e53f049f04fe9c889fb20a4fe6d173L35-R35): Updated log messages to reflect the change from "inference" to "prediction".
* [`.ci/tests/examples/run_prediction.sh`](diffhunk://#diff-fd5e99a3c77cd4b821b5e0dd903a21c9340e2dc40f312d79f92729b80f81a6d1L11-R16): Renamed script and updated curl commands and log messages to use "prediction" instead of "inference".
* [`examples/mnist-keras/client/predict.py`](diffhunk://#diff-229f79748ca1241afee776aca70032026f819833741915479e9356c2b716ea93L14-R20): Changed comments and method calls from "inference" to "prediction".
* [`fedn/network/api/v1/__init__.py`](diffhunk://#diff-c74255f57cb1e530964fe3aa7855e5b53e7af90f36ffccc52a856cbd1987bdb7L3-R11): Updated route imports to use `prediction_routes` instead of `inference_routes`.
* [`fedn/network/api/v1/prediction_routes.py`](diffhunk://#diff-ddca183ab51938a308d2906154a96a51e1a25537556774f8eb95787407053188R1-R37): Added new prediction routes and removed the old inference routes.

### API Changes

* [`fedn/network/clients/client.py`](diffhunk://#diff-04f9a0c7d1dffb7bbe3e42520f43ce47af964c4eeacdf7a32eca505ece1e79e0L421-R425): Refactored methods to handle prediction requests instead of inference requests, including logging and status updates. [[1]](diffhunk://#diff-04f9a0c7d1dffb7bbe3e42520f43ce47af964c4eeacdf7a32eca505ece1e79e0L421-R425) [[2]](diffhunk://#diff-04f9a0c7d1dffb7bbe3e42520f43ce47af964c4eeacdf7a32eca505ece1e79e0L522-R532) [[3]](diffhunk://#diff-04f9a0c7d1dffb7bbe3e42520f43ce47af964c4eeacdf7a32eca505ece1e79e0L553-R545) [[4]](diffhunk://#diff-04f9a0c7d1dffb7bbe3e42520f43ce47af964c4eeacdf7a32eca505ece1e79e0L569-R576) [[5]](diffhunk://#diff-04f9a0c7d1dffb7bbe3e42520f43ce47af964c4eeacdf7a32eca505ece1e79e0L594-R599) [[6]](diffhunk://#diff-04f9a0c7d1dffb7bbe3e42520f43ce47af964c4eeacdf7a32eca505ece1e79e0L671-R663) [[7]](diffhunk://#diff-04f9a0c7d1dffb7bbe3e42520f43ce47af964c4eeacdf7a32eca505ece1e79e0L710-R742)
* [`fedn/network/combiner/combiner.py`](diffhunk://#diff-ef69b399917218afadbf6931e5791eac521bcb7623789475cf0437d32534e31eR12-R19): Updated client handling and status management to reflect changes in prediction handling. [[1]](diffhunk://#diff-ef69b399917218afadbf6931e5791eac521bcb7623789475cf0437d32534e31eR12-R19) [[2]](diffhunk://#diff-ef69b399917218afadbf6931e5791eac521bcb7623789475cf0437d32534e31eR87-R92) [[3]](diffhunk://#diff-ef69b399917218afadbf6931e5791eac521bcb7623789475cf0437d32534e31eR117-R139)

### Minor Improvements and Bug Fixes

* [`fedn/cli/status_cmd.py`](diffhunk://#diff-1d18776ec6d976b363157de5095c3e110a94b835c24521e366ac012d2a7e095eL11-R11): Cleaned up docstring formatting.
* [`fedn/network/api/v1/status_routes.py`](diffhunk://#diff-e9c41d5efca071cd09d2603d4f9f82ad5e165395b1066d7197b21a40ee33473aR127-R132): Added debug prints for typed headers and request arguments.
* [`.github/workflows/integration-tests.yaml`](diffhunk://#diff-213e9bbc551ccc9ca3cbb6b69999a415116e2ee37ccc88e34a98c5dd27bbc31aL41-R42): Updated commented-out lines to reflect the change from "inference" to "prediction".
* [`fedn/network/clients/state.py`](diffhunk://#diff-8fc9b581e379b8fa57bdbf62b335e1272f189b2f37671d6c580598523a985191L10-R10): Updated `ClientState` enum to replace "inferencing" with "predicting".

These changes ensure consistency in terminology and improve the clarity and functionality of the codebase.

### CODE REVIEW

**Review by ahellander — COMMENTED** [review]
Looks good! But should we not call them "ModelPrediction" ?

File: fedn/network/clients/client.py [hunk 1]
```diff
@@ -566,22 +558,22 @@ def _process_validation_request(self, model_id: str, is_inference: bool, session
         self.state = ClientState.idle
         return validation
 
-    def _process_inference_request(self, model_id: str, session_id: str, presigned_url: str):
-        """Process an inference request.
+    def _process_prediction_request(self, model_id: str, session_id: str, presigned_url: str):
+        """Process an prediction request.
```
Comment (Wrede) [inline fedn/network/clients/client.py:562]: "a" predtiction

File: fedn/network/clients/state.py [hunk 2]
```diff
@@ -7,7 +7,7 @@ class ClientState(Enum):
     idle = 1
     training = 2
     validating = 3
-    inferencing = 4
+    predicting = 4
```
Comment (Wrede) [inline fedn/network/clients/state.py:10]: We can also add the ClientStateToString for "predicting"

File: fedn/network/clients/client.py [hunk 3]
```diff
@@ -707,21 +699,47 @@ def process_request(self):
...
+                    # TODO: Add prediction data
+                    prediction.data = ""
+                    prediction.timestamp.GetCurrentTime()
+                    prediction.correlation_id = request.correlation_id
+                    # Obs that session_id in request is the prediction_id
+                    prediction.prediction_id = request.session_id
+
+                    try:
+                        _ = self.combinerStub.SendModelPrediction(prediction, metadata=self.metadata)
+                        status_type = fedn.StatusType.INFERENCE
```
Comment (Wrede) [inline fedn/network/clients/client.py:734]: We can change StatusType.INFERENCE to StatusType.MODEL_PREDICTION in fedn.proto

File: fedn/network/combiner/combiner.py [hunk 4]
```diff
@@ -191,29 +209,31 @@ def request_model_validation(self, session_id, model_id, clients=[]):
...
+        :param model_id: the model id to perform prediction on
         :type model_id: str
         :param config: the model configuration to send to clients
         :type config: dict
         :param clients: the clients to send the request to
         :type clients: list
 
         """
-        clients = self._send_request_type(fedn.StatusType.INFERENCE, session_id, model_id, clients)
+        clients = self._send_request_type(fedn.StatusType.INFERENCE, prediction_id, model_id, clients)
```
Comment (Wrede) [inline fedn/network/combiner/combiner.py:223]: see other comment about StatusType

**Review by Wrede — COMMENTED** [review]
had some comments

**Review by ahellander — APPROVED** [review]

**Pull request merged**